src/training/ea/lipizzaner_gan_trainer.py

```diff
@@ -214,12 +235,98 @@ def train(self, n_iterations, stop_event=None):
                                            self.score, stop_time - start_time,
                                            path_real_images, path_fake_images)
 
+        if self.optimize_weights_at_the_end:
+            self.optimize_generator_mixture_weights()
+
+            path_real_images, path_fake_images = \
+                self.log_results(batch_size, iteration+1, input_data, loaded,
+                                 lr_gen=self.concurrent_populations.generator.individuals[0].learning_rate,
+                                 lr_dis=self.concurrent_populations.discriminator.individuals[0].learning_rate,
+                                 score=self.score, mixture_gen=self.neighbourhood.mixture_weights_generators,
+                                 mixture_dis=self.neighbourhood.mixture_weights_discriminators)
+
+            if self.db_logger.is_enabled:
+                self.db_logger.log_results(iteration+1, self.neighbourhood, self.concurrent_populations,
+                                           self.score, stop_time - start_time,
+                                           path_real_images, path_fake_images)
+
+
         return self.result()
 
+    def optimize_generator_mixture_weights(self):
```

Comment:
Is there a test for this function? Seems useful to even call standalone, then a test can help to show how to do this, and check that it is running